specs/Print.md

```diff
@@ -0,0 +1,843 @@
+  /// for descriptions of print dialog kinds.
+  ///
+  /// \snippet AppWindow::ShowPrintUI
+  HRESULT ShowPrintUI([in] COREWEBVIEW2_PRINT_DIALOG_KIND printDialogKind);
+
+  /// Provides the Pdf data of current web page asynchronously for the provided settings.
+  /// See `ICoreWebView2PrintSettings2` for description of settings. Passing
+  /// nullptr for `printSettings` results in default print settings used.
+  ///
+  /// The async `PrintToPdfStream` operation completes when it finishes
+  /// writing to the stream. At this time the `ICoreWebView2PrintToPdfStreamCompletedHandler`
+  /// is invoked. Only one `Printing` operation can be in progress at a time. If
+  /// `PrintToPdfStream` is called while a `PrintToPdfStream` or `PrintToPdf` or `Print` or `ShowPrintUI`
+  /// job is in progress, the completed handler is immediately invoked with `E_ABORT`.
+  ///
+  /// \snippet AppWindow.cpp PrintToPdfStream
+  HRESULT PrintToPdfStream([in] ICoreWebView2PrintSettings2* printSettings,
+                           [in] ICoreWebView2PrintToPdfStreamCompletedHandler* handler);
+}
+
+/// Receives the result of the `Print` method.
+/// |       errorCode     |       isSuccessful    |               Notes                                                                           |
+/// |        S_OK         |        True           | Print operation succeeded.                                                                    |
+/// |        S_OK         |        False          | If specified printer is not found or printer status is not available, offline or error state. |
+/// |       E_FAIL        |        False          | Print operation is failed.                                                                    |
+/// |     E_INVALIDARG    |        False          | If the caller provides invalid settings for the specified printer.                            |
+/// |       E_ABORT       |        False          | Print operation is failed as printing job is in progress.                                     |
+[uuid(4B2FE043-1A1A-4924-803D-C05E5FDCA3B0), object, pointer_default(unique)]
+interface ICoreWebView2PrintCompletedHandler : IUnknown {
+  /// Provides the result of the corresponding asynchronous method.
+  HRESULT Invoke([in] HRESULT errorCode, [in] BOOL isSuccessful);
+}
+
+/// Receives the result of the `PrintToPdfStream` method.
+/// `errorCode` returns S_OK if the PrintToPdfStream operation succeeded.
+/// The printable pdf data is returned in the `pdfStream` object.
+[uuid(78A9FC37-249B-4E7E-A2E6-20CAAA14AE7B), object, pointer_default(unique)]
+interface ICoreWebView2PrintToPdfStreamCompletedHandler : IUnknown {
+  /// Provides the result of the corresponding asynchronous method.
+  HRESULT Invoke([in] HRESULT errorCode, [in] IStream* pdfStream);
+}
+
+/// Settings used by the `Print` method.
+[uuid(DA3AD25C-D121-44DC-83DA-BF5E50DBF0DF), object, pointer_default(unique)]
+interface ICoreWebView2PrintSettings2 : ICoreWebView2PrintSettings {
```

Comment:
Following PrintSettings2 are changed:

UWP WinRT - Collation has these enum values Collated, Default, NotAvailable ,PrinterCustom, Uncollated.

WebView2 API
`ShouldCollate` (bool) to `Collation` (COREWEBVIEW2_PRINT_COLLATION) with enum values Default, Collated, Uncollated. 

Chromium code supports two values (collated/not collated). Default is the user configured default collation for printer.

UWP WinRT - ColorMode has these enum values Color, Default, GrayScale.

WebView2 API
`IsColor` (bool) to `ColorMode` (COREWEBVIEW2_PRINT_COLOR_MODE) with enum values Default, Color, Grayscale.

Chromium code supports two values (Color/Grayscale). Default is the user configured default color for printer.

UWP WinRT - PrintDuplex (Default, NotAvailable, OneSided, PrinterCustom, TwoSidedLongEdge, TwoSidedShortEdge)

WebView2 API
PrintSide (COREWEBVIEW2_PRINT_SIDE) to Duplex (COREWEBVIEW2_PRINT_DUPLEX) with values (Default, OneSided, TwoSidedLongEdge, TwoSidedShortEdge) that matches with chromium.

UWP WinRT - 
PrintMediaSize (IsoA0 , IsoA1, IsoA02..............................)

WebView2 API
Added MediaSize (COREWEBVIEW2_PRINT_MEDIA_SIZE) with values (Default and Custom). If custom user has to specify PageWidth and PageHeight. If default, it prints to user configured default paper for the printer.

We removed QualityHorizontal and QualityVertical in our API

UWP WinRT API `PrintQuality` has (Automatic, Default, Draft, Fax, High, Normal, NotAvailable, Photographic, PrinterCustom, Text).

Chromium code takes quality in dpi value and there isn't an easy way to determine high/draft etc.. values in dpi for the printer. 

Also, there is no customer request for this property. We plan to investigate and add in future if needed.

Reply:
Great!